api/v1beta1/azuremanagedmachinepool_webhook_test.go

```diff
@@ -1220,6 +1221,7 @@ func TestAzureManagedMachinePool_validateLastSystemNodePool(t *testing.T) {
 					Name:              systemMachinePool.GetLabels()[clusterv1.ClusterNameLabel],
 					Namespace:         systemMachinePool.Namespace,
 					DeletionTimestamp: &deletionTime,
+					Finalizers:        finalizers,
```

Comment:
Why are these Finalizers required? I don't see it being used in other tests that are refactored.

Reply:
Good question. This was hard to track down, but some unit tests started erroring with this change, and it boiled down to this (from the controller-runtime v0.15.0 [release notes](https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.15.0)):
> Update fake client deletionTimestamp behavior (https://github.com/kubernetes-sigs/controller-runtime/pull/2316)
>  - The fake client will panic if initialized with an object that has a DeletionTimestamp and no finalizer